content/docs/migration_1.0/C++.md

```diff
@@ -0,0 +1,363 @@
+  
+  // arithmetic type serialization
+  double pi = 3.1415926;
+  Bytes b = Bytes::serialize(pi);
+  assert(b.deserialize<TYPE>() == pi);
+  
+  // Composite types serialization
+  std::vector<float> v = {0.1f, .2f, 0.3f};
+  auto b = Bytes::serialize(v);
+  assert(b.deserialize<decltype(v)>() == v);
+  
+  std::map<std::string, std::deque<double>> m = {
+    {"a", {0.5, 0.2}},
+    {"b", {-123.45, 0.4}},
+    {"abc", {3.1415926, -1.0} }
+  };
+
+  b = Bytes::serialize(m);
+  assert(b.deserialize<decltype(m2)>() == m); 
+
+  // alternatively serialize via move 
+  // the string keys will not be copied in this case, but rather move into Bytes
+  b_move = Bytes::serialize(std::move(m));
+```
+
+Please note that this serialization functionality is only provided for prototyping and demonstration purposes and, as such, might be less efficient than custom-written serialization methods.
+
+Users can easily define their own serialization/deserialization functions by providing a custom codec:
+
+```cpp
+MyType my_data(...);
+MyCodec my_codec(...);
+Bytes b = Bytes::serialize(my_data, my_codec);
+// or Bytes::serialize<MyCodec>(my_data) if MyCodec is a stateless codec with an empty constructor 
+assert(b.deserialize<std::vector<MyType>>(my_codec) == my_data);
+// or assert(b.deserialize<std::vector<MyType, MyCodec>>() == my_data); if MyCodec is a stateless with an empty constructor
+```
+
+For finer control on serialization / deserialization implementation `Bytes::Iterator`, `Bytes::Writer` and `Bytes::Reader` classes are also introduced.
+
+A working example with custom-defined serialization / deserialization can be found here:
+
+https://github.com/eclipse-zenoh/zenoh-cpp/blob/dev/1.0.0/examples/simple/universal/z_simple.cxx
+
+## Stream Handlers and Callbacks
+
+Prior to 1.0.0, stream handlers were only supported for `get` :
+
+```cpp
+// callback
+session.get(keyexpr, "", {on_reply, on_done}, opts);
+
+// stream handlers interface
+auto [send, recv] = reply_fifo_new(16);
+session.get(keyexpr, "", std::move(send), opts);
+
+Reply reply(nullptr);
+// blocking
+for (recv(reply); reply.check(); recv(reply)) {
+    auto sample = expect<Sample>(reply.get());
+    std::cout << "Received ('" << sample.get_keyexpr().as_string_view() << "' : '"
+              << sample.get_payload().as_string_view() << "')\n";
+}
+
+// non-blocking
+for (bool call_success = recv(reply); !call_success || reply.check(); call_success = recv(reply)) {
+  if (!call_success) {
+    std::cout << ".";
+    Sleep(1);
+    continue;
+  }
+  auto sample = expect<Sample>(reply.get());
+  std::cout << "\nReceived ('" << sample.get_keyexpr().as_string_view() << "' : '"
+            << sample.get_payload().as_string_view() << "')";
+}
+```
+
+In 1.0.0, `Subscriber`, `Queryable` and `get` can now use either a callable object or a stream handler. Currently, Zenoh provides 2 types of handlers:
+
+- `FifoHandler` - serving messages in Fifo order, *dropping new messages* once full.
```

Comment:
@DenisBiryukov91 does the `FifoHandler` drops message? It should NOT drop messages but rather block.

Reply:
No it does not, it blocks as expected. I guess it comes from using outdated version of migration guide in this pr.

Reply:
updated